tests/test_methods_and_attributes.py

```diff
@@ -171,6 +171,13 @@ def test_static_properties():
     assert m.TestPropertiesOverride().def_readonly == 99
     assert m.TestPropertiesOverride.def_readonly_static == 99
 
+    # Static attributes can be deleted
+    del m.TestPropertiesOverride.def_readonly_static
+    assert not hasattr(m.TestPropertiesOverride, "def_readonly_static")
```

Comment:
So this check fails because:

```
>>> import pybind11_tests
>>> m = pybind11_tests.methods_and_attributes
>>> m.TestPropertiesOverride.def_readonly_static
99
>>> del m.TestPropertiesOverride.def_readonly_static
>>> hasattr(m.TestPropertiesOverride, 'def_readonly_static')
True
>>> m.TestPropertiesOverride.def_readonly_static
1
>>> del m.TestPropertiesOverride.def_readonly_static
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: def_readonly_static
```

Can anyone make sense of it?

Reply:
The latter value, 1, seems to come from `m.TestProperties`, a base class of `m.TestPropertiesOverride`?

Reply:
That's right.

```
>>> m = pybind11_tests.methods_and_attributes
>>> del m.TestPropertiesOverride.def_readonly_static
>>> m.TestPropertiesOverride.def_readonly_static
1
>>> del m.TestProperties.def_readonly_static
>>> m.TestPropertiesOverride.def_readonly_static
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'pybind11_tests.methods_and_attributes.TestProperti' has no attribute 'def_readonly_static'
```

This makes me even less sure if this is the right course of action.

Reply:
Can we see/mimic how this would react in Python?

Reply:
Just tried this out:

```pycon
>>> class Base:
...     answer = 6 * 9
... 
>>> class Derived(Base):
...     answer = 42
... 
>>> Derived.answer
42
>>> del Derived.answer
>>> Derived.answer
54
>>> del Derived.answer
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: answer
>>> del Base.answer
>>> Derived.answer
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'Derived' has no attribute 'answer'
>>> 
```

So even without any properties, static attributes in pure Python would match our behavior, no?

So ... not a bug, just counter-intuitive Python behaviour?